apps/total-typescript/src/templates/workshop-template.tsx

```diff
@@ -214,8 +244,13 @@ const WorkshopSectionNavigator: React.FC<{workshop: Module}> = ({workshop}) => {
 const SectionItem: React.FC<{
   section: Section
   workshop: Module
-}> = ({section, workshop}) => {
-  const {isCompleted, percentCompleted} = useSectionProgress({section: section})
+  moduleProgress: ModuleProgress | null | undefined
+}> = ({section, workshop, moduleProgress}) => {
```

Comment:
same here, instead of props just call the router. it caches and batches so it will optimize under the hood

Reply:
very cool! will fix